pyproject.toml

```diff
@@ -0,0 +1,40 @@
+[build-system]
+requires = ["flit >=3.2,<4"]
+build-backend = "flit.buildapi"
+
+[project]
+name = "spark_log_parser"
+authors = [{"name" = "Sync Computing"}]
+readme = "README.md"
+requires-python = ">=3.6.1"
```

Comment:
Would it be a good idea to fix the version to the same runtime.txt as in sync_backend? Not sure if there's any risk with having different spark versions operating on the backend and the log_parser.

Reply:
I found it hard to justify narrowing the supported Python versions here though I wanted to. I'm not aware of anything in this repo that demands a higher version now, and having a wider range of versions might make it easier for users. On the other hand, there's probably some risk of introducing something incompatible with sync_backend and not finding out until we upgrade the dependency there but that's true with any dependency.

Reply:
Ok. For the external client use-case the risk is very low, since __main__.py just saves a json dump. We'll just have to be wary when updates are made that whatever is being introduced doesn't break compatibility for the lower end of the versions here.

Reply:
Ended up changing it to ">=3.10.3" in a subsequent PR so that contributions can use the newer type annotations.